subprojects/core/src/test/groovy/org/gradle/execution/plan/DefaultExecutionPlanParallelTest.groovy

```diff
@@ -263,21 +264,25 @@ class DefaultExecutionPlanParallelTest extends AbstractExecutionPlanSpec {
         tasksAreNotExecutedInParallel(a, b)
     }
 
+    @ToBeImplemented("When we support symlinks in the VFS, we should implement this as well")
     @Requires(TestPrecondition.SYMLINKS)
     def "a task that writes into a symlink that overlaps with output of currently running task is not started"() {
         given:
         def taskOutput = file("outputDir").createDir()
         def symlink = file("symlink")
-        fs.createSymbolicLink(symlink, taskOutput)
+        symlink.createLink(taskOutput)
 
         and:
         Task a = task("a", type: AsyncWithOutputDirectory)
         _ * a.outputDirectory >> taskOutput
         Task b = task("b", type: AsyncWithOutputFile)
-        _ * b.outputFile >> symlink.file("fileUnderSymlink")
+        _ * b.outputFile >> new File(symlink, "fileUnderSymlink")
+
+        addToGraphAndPopulate(a, b)
 
         expect:
-        tasksAreNotExecutedInParallel(a, b)
+        // TODO: Should be tasksAreNotExecutedInParallel(a, b)
+        [selectNextTask(), selectNextTask()] as Set == ([a, b] as Set)
```

Comment:
Nit: shall we use `when:` and `then:` here? Like, `when: addToGraphAndPopulate(a, b); then: ...` sounds better to me.

Reply:
I added a method called `tasksAreExecutedInParallel()`, so we are closer to the original test code and I use that method now in `expect`.